Symbiota/Symbiota
# [3.3] 1924 export associations as resource relationship

# Description

This PR allows for the download of related records. It addresses issue #1924 .

From the issue specs:

> Adds DwcArchiverResourceRelationship class (following format similar to DwcArchiverIdentifier, DwcArchiverMaterialSample, and DwcArchiverAttribute
Add include Resource Relationship extension checkbox to all download interfaces and web services
Currently External and Internal Associations are concatenated into associatedOccurrences within occurrences.csv core file.
associatedOccurrences will be removed from DwC-Archive export (simply comment out lines 137 and 138 within DwcArchiverOccurrence). We might remove this column from the omoccurrences schema, but let's just comment it out for now.
Within DwcArchiverOccurrence getAssociationStr and getAssociationJSON will be completely removed.
appendSpecimenDuplicateAssociations function will be removed with functionality integrated as a batch function, similar to functions called between lines 1821 and 1827 within DwcArchiverCore
Currently observation associations are appended into the associatedTaxa field, which needs to be removed. The associatedTaxa field will remain in the DwC Archive export, but should now only contain verbatim text from the associatedTaxa field.

# QA Notes

Live on dev001. In addition to all of the nuances of what we should map from Symbiota Native fields to DwC, note that I haven't tested the Darwin Core-only download or skipping the download entirely in a while.

# Notes/Known Issues

This PR does not accommodate external or observational associations. They are in a [separate issue](https://github.com/BioKIC/Symbiota/issues/2091).

# Pull Request Checklist:

# Pre-Approval

- [x] There is a description section in the pull request that details what the proposed changes do. It can be very brief if need be, but it ought to exist.
- [ ] Hotfixes should be branched off of the `master` branch and PR'd using the **merge** option (not squashed) into the `hotfix` branch.
    - N/A
- [x] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
- [ ] All new text is preferably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages), and the [spreadsheet tracking internationalizations](https://docs.google.com/spreadsheets/d/133fps9w2pUCEjUA6IGCcQotk7dn9KvepMXJ2IWUZsE8/edit?usp=sharing) has been updated either with a new row or with checkmarks to existing rows.
    - N/A
- [ ] New features have responsive design (i.e., look aesthetically pleasing both full screen and with small or mobile screens)
    - N/A
- [x] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [x] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
    - If so, noted in code diff
- [x] Comment which GitHub issue(s), if any does this PR address
    - [https://github.com/BioKIC/Symbiota/issues/1924](https://github.com/BioKIC/Symbiota/issues/1924)
- [ ] If this PR makes any changes that would require additional configuration of any Symbiota portals outside of the files tracked in this repository, make sure that those changes are detailed in [this document](https://docs.google.com/document/d/1T7xbXEf2bjjm-PMrlXpUBa69aTMAIROPXVqJqa2ow_I/edit?usp=sharing).
    - N/A

# Post-Approval

- [ ] It is the code author's responsibility to merge their own pull request after it has been approved
- [ ] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [ ] If this PR represents a merge into the `hotfix` branch, remember to use the **merge** option (i.e., no squash).
- [ ] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option
- [ ] If this PR represents a merge from the `hotfix` branch into the `master` branch use the **squash & merge** option
  - [ ] a subsequent PR from `master` into `Development` should be made with the **merge** option (i.e., no squash).
  - [ ] **Immediately** delete the `hotfix` branch and create a new `hotfix` branch
  - [ ] increment the Symbiota version number in the symbase.php file and commit to the `hotfix` branch.
- [ ] If the dev team has agreed that this PR represents the last PR going into the `Development` branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place. Follow the release protocol [here](https://github.com/BioKIC/Symbiota/blob/master/docs/release-protocol.md).
- [ ] Don't forget to delete your feature branch upon merge. Ignore this step as required.

Thanks for contributing and keeping it clean!
